core/src/main/scala/kafka/server/ClientMetricsManager.scala

```diff
@@ -17,37 +37,73 @@ import scala.jdk.CollectionConverters._
 object ClientMetricsManager {
   private val _instance = new ClientMetricsManager
   def getInstance = _instance
+  def getCurrentTime = Calendar.getInstance.getTime.getTime
 
   def getSupportedCompressionTypes: List[java.lang.Byte] = {
     val compressionTypes = new ListBuffer[java.lang.Byte]
-    CompressionType.values.foreach(x => compressionTypes.append(x.id.toByte))
+    CompressionType.values.filter(x => x != CompressionType.NONE).foreach(x => compressionTypes.append(x.id.toByte))
     compressionTypes.toList
   }
 
+  // TODO: Needs to integrate with external plugin changes..
+  // if plugin is not configured, getMetricsSubscriptions and pushMetricSubscription needs to return errors
+  // to the client.
```

Comment:
I'm not sure this is the correct behavior.

If there are no plugins, the broker should probably disable telemetry via an "empty" subscription.

But this does bring up a good point -- how do we totally disable telemetry push and get subscriptions calls from the broker side?

Reply:
Not sure whatz the right behavior is but what good broker can do if there is no configured plugin to process the metrics data.

Reply:
I agree that it's pointless to send telemetry up to the broker if the broker isn't configured to do anything with it.

Throwing an exception on metric subscription retrieval and/or push telemetry will just cause the client to retry the get telemetry subscription call again. I guess the error case on get telemetry subscription is relatively lightweight. I think I'm mostly objecting to the connotation of "error" vs. "disabled". The client isn't doing "wrong" in asking for the telemetry, so receiving an "error" seems to imply it has.

I think I'm arguing for something that implies "disabled" vs. "error".

The [KIP section on compatibility](https://cwiki.apache.org/confluence/display/KAFKA/KIP-714%3A+Client+metrics+and+observability#KIP714:Clientmetricsandobservability-Broker) says:

> If no such plugin is configured the telemetry collection is disabled and there will be no user-facing changes.

The KIP doesn't define the semantics when the plugin isn't configured. Might be worth updating the KIP with a sentence or two. Or perhaps we need to add that to the section for error codes.

Reply:
I'm happy to work on fleshing out how to communicate that the telemetry is disabled. LMK what you think.

Reply:
https://confluentinc.atlassian.net/browse/KCFUN-360